go/applychanges/apply_changes_test.go

```diff
@@ -135,3 +135,43 @@ func TestApplyChangesUUID(t *testing.T) {
 	assert.Equal(t, "02a9920e-b015-4de3-8e32-fb965ac4653c", clay.ID.String())
 	assert.Equal(t, "Clay", clay.Name)
 }
+
+type EmbedStruct struct {
+	ID uuid.UUID `json:"id"`
+}
+type Embedder struct {
+	EmbedStruct
+	Name string `json:"name"`
+}
+type UltimateEmbedder struct {
+	Embedder
+	PowerLevel int `json:"powerLevel"`
+}
+
+func TestApplyChangesOnEmbeddedStructs(t *testing.T) {
+	champion := UltimateEmbedder{
+		PowerLevel: 7900,
+		Embedder: Embedder{
+			Name: "Vegeta",
+			EmbedStruct: EmbedStruct{
+				ID: uuid.Nil,
+			},
+		},
+	}
+
+	happyChanges := map[string]interface{}{
+		"id":         []byte{0x2, 0xa9, 0x92, 0xe, 0xb0, 0x15, 0x4d, 0xe3, 0x8e, 0x32, 0xfb, 0x96, 0x5a, 0xc4, 0x65, 0x3c},
```

Comment:
Rather than a hard-to-read byte sequence, could we hardcode the UUID as a string, than marshal it into bytes when needed? Something like this:
```go
newID := "02a9920e-b015-4de3-8e32-fb965ac4653c"
newIDBytes, err := uuid.MustParse(newID).MarshalBinary()
assert.NoError(t, err)

happyChanges := map[string]interface{}{
	"id":         newIDBytes,
	"powerLevel": 9001,
	"name":       "Goku",
}
```
and then the assertion below can be
```go
assert.Equal(t, newID, champion.ID.String())
```

Reply:
I like this a lot! I think we should implement this.

Reply:
I think that is a good call! I don't think I even need to test applying the bytes here, as that was being tested in an earlier test.  I simplified this to 

```go
	newID := "02a9920e-b015-4de3-8e32-fb965ac4653c"

	happyChanges := map[string]interface{}{
		"id":         uuid.MustParse(newID),
		"powerLevel": 9001,
		"name":       "Goku",
	}
```

Reply:
With https://github.com/CMSgov/easi-shared/pull/13/commits/69cb880fe21db65a750fd18913aae0cb0ce9ddea, if the test works fine here applying a `uuid.UUID`, we can probably remove the marshaling to a `[]byte` in `TestApplyChangesUUID()` as well.

Reply:
@ClayBenson94 Do you know why the test in `TestApplyChangesUUID()` uses a `[]byte`?

Reply:
I left it as a `[]byte`, just so it is a test that shows we can use ApplyChanges with a `[]byte`, since it is also using `uuid.MustParse(newID)` in the new unit test. I could go either way removing it or keeping it, but I thought it wouldn't hurt to have another diverse test here since it is a shared component.

Reply:
I'm thinking when I wrote the test I just didn't realize that `applyChanges` would work natively with a `uuid` type. I think the most important tests are the ones that reflect (hah hah!) what we're actually passing to applychanges. If we're passing `uuid` types, then I think this test as-is makes sense! (Which we probably are? I think because we have a custom [UUID type for GQLGen](https://github.com/CMSgov/mint-app/blob/bc7b437f7aef9490661467fe1f78cd6dbe54e3d6/pkg/models/uuid.go#L1-L22) we're probably passing them there)

Reply:
Sounds fine to me as it is.